keps/sig-cli/NNNN-kustomize-components.md

```diff
@@ -0,0 +1,676 @@
+---
+title: Kustomize Components
+authors:
+  - "@apyrgio"
+  - "@ioandr"
+  - "@pgpx"
+owning-sig: sig-cli
+participating-sigs:
+  - n/a
+reviewers:
+  - n/a
+approvers:
+  - n/a
+creation-date: 2020-05-??
+last-updated: 2020-05-??
+status: provisional
+see-also:
+  - n/a
+replaces:
+  - n/a
+superseded-by:
+  - n/a
+---
+
+# Kustomize Components
+
+<!-- toc -->
+- [Release Signoff Checklist](#release-signoff-checklist)
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Story](#user-story)
+  - [Notes/Constraints/Caveats (optional)](#notesconstraintscaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+(**FIXME:** Explain what is the current situation and then roughly explain what this feature does)
+
+> This section is incredibly important for producing high quality user-focused
+> documentation such as release notes or a development roadmap.  It should be
+> possible to collect this information before implementation begins in order to
+> avoid requiring implementors to split their attention between writing release
+> notes and implementing the feature itself.  KEP editors, SIG Docs, and SIG PM
+> should help to ensure that the tone and content of the `Summary` section is
+> useful for a wide audience.
+>
+> A good summary is probably at least a paragraph in length.
+
+## Motivation
+
+By design, Kustomize and its variant system follow an inheritance model rather
+than a composition one. This poses certain limitations in complex deployment
```

Comment:
I don't think we want this to be a this-or-that situation, so I'd remove the "rather than a composition one" (which we haven't explained at this point yet)

Reply:
ACK, makes sense.